### PR TITLE
Add monitor options for synthetics options

### PR DESCRIPTION
**Pull request by KeisukeYamashita**
## What

* Add `MonitorOptions' for Synthetic test

## Why

* There is `.options.monitor_options` .
   * `renotify_interval` is very important.

With current API you can create a like below.

```
curl -X "POST" "https://api.datadoghq.com/api/v1/synthetics/tests" \
     -H 'Content-type: application/json' \
     -H 'DD-API-KEY:xxxxxx' \
     -H 'DD-APPLICATION-KEY: xxxxxxxx' \
     -d $'{
  "config": {
    "assertions": [
      {
        "type": "certificate",
        "operator": "isInMoreThan",
        "target": 45
      }
    ],
    "request": {
      "host": "example.com",
      "port": 443
    }
  },
  "message": "[Keke test] KeisukeYamashita",
  "locations": [
    "aws:ap-northeast-1"
  ],
  "subtype": "ssl",
  "type": "api",
  "options": {
    "tick_every": 60,
    "monitor_options": {
      "renotify_interval": 200
    }
  },
  "tags": [
    "env:production",
    "team:merpay-sre",
    "user:keisukeyamashita",
    "host:example.com"
  ],
  "name": "[Keke test] An KeisukeYamashita test"
}'
```

### CODE REVIEW

File: synthetics.go [hunk 1]
```diff
@@ -52,12 +52,17 @@ type SyntheticsAssertion struct {
...
-	MinLocationFailed  *int     `json:"min_location_failed,omitempty"`
-	DeviceIds          []string `json:"device_ids,omitempty"`
-	AcceptSelfSigned   *bool    `json:"accept_self_signed,omitempty"`
+	TickEvery          *int           `json:"tick_every,omitempty"`
+	FollowRedirects    *bool          `json:"follow_redirects,omitempty"`
+	MinFailureDuration *int           `json:"min_failure_duration,omitempty"`
+	MinLocationFailed  *int           `json:"min_location_failed,omitempty"`
+	DeviceIds          []string       `json:"device_ids,omitempty"`
+	AcceptSelfSigned   *bool          `json:"accept_self_signed,omitempty"`
+	MonitorOptions     MonitorOptions `json:"monitor_options,omitempty"`
```
Comment (btoueg) [inline synthetics.go]: If `MonitorOptions` is not a pointer, bad things can happen (see https://play.golang.org/p/c3vXmQO6GeH)

```suggestion
	MonitorOptions     *MonitorOptions `json:"monitor_options,omitempty"
```

Comment (KeisukeYamashita) [reply synthetics.go]: Thank you very much. Yes, `nil` ain't good. Thank you very much.

**Comment by KeisukeYamashita** [comment]
@btoueg  Thanks for your review. I've added the missing test.

**Comment by KeisukeYamashita** [comment]
@btoueg  Please review. Thank you in advance.

**Comment by KeisukeYamashita** [comment]
@btoueg  Thank you for your review. I've fixed it. Please take a look.

**Pull request merged**